### ISSUE TITLE
m2090B0A12.flver can't be exported due to having 92908 vertices

### ISSUE THREAD

**Issue by fromsoftserve** [Grimrukh/soulstruct-blender#56]
And changing the normal/tangent value doesn't get it below the necessary vertices to export. And this is without modifying the model at all. All I did was turn on the smooth mesh shading, but no model edits. 
![toomanyvertices](https://github.com/user-attachments/assets/732ee742-59f9-4b94-8047-91650d76387f)

**Comment by Grimrukh** [comment]
tl;dr: this isn't a bug and you'll either need to lower Normal Tangent Dot Max in the FLVER export settings (which will "undo" some of the smooth shading) or manually split the submesh to get under the hard vertex limit.

If you turn on smooth mesh shading, Blender will generate separate "split normals" at every vertex for every face around it. Each of these split normals then needs to be written to the FLVER as a separate vertex, because FLVER files are optimized for rendering and cannot store variable "face corner" data, like split normals, without duplicating everything about a vertex. If you leave Normal Tangent Dot Max at 1, Soulstruct will do zero merging of split normals, and you are guaranteed to blow out the number of vertices with smooth shading (which is why they didn't do it in 2011). By lowering it, you will basically be gradually undoing the smooth shading enabled in Blender, but only in cases where the normals are already very similar (depending on how low it gets).

Of course, in DSR, the FLVER submesh vertex limit doesn't reflect any real performance concern anymore and is just a stupid artifact of the past. You could circumvent this issue entirely by manually dividing the large submesh into two or more submeshes, by duplicating its material. Every Blender material will be exported by Soulstruct as a different submesh, even if the material data is identical.

Or, final option: under FLVER > FLVER Model Tools > Mesh Tools, you will see an operator called "Set Smooth Custom Normals". I created this for exactly this purpose: getting nice smooth normals while maintaining only one per vertex. However, this is designed for new geometry added, not as a way to upgrade existing geometry, as it intentionally mimics the original FLVER optimization. But you could select certain parts of the mesh in Edit Mode that don't benefit as much from global smooth shading, and then use this operator. (Or press Alt + N to see all the different ways Blender can manipulate normals and come up with your own compromise.)
